presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleTTLNodeSelector.java

```diff
@@ -0,0 +1,264 @@
+import static com.facebook.presto.execution.scheduler.NodeScheduler.toWhenHasSplitQueueSpaceFuture;
+import static com.facebook.presto.spi.StandardErrorCode.NODE_SELECTION_NOT_SUPPORTED;
+import static com.facebook.presto.spi.StandardErrorCode.NO_NODES_AVAILABLE;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+public class SimpleTTLNodeSelector
+        implements NodeSelector
+{
+    private static final Logger log = Logger.get(SimpleTTLNodeSelector.class);
+    private final TTLFetcherManager ttlFetcherManager;
+    private final Session session;
+    private final AtomicReference<Supplier<NodeMap>> nodeMap;
+    private final NodeTaskMap nodeTaskMap;
+    private final int minCandidates;
+    private final int maxSplitsPerNode;
+    private final int maxPendingSplitsPerTask;
+    private final int maxTasksPerStage;
+    private final boolean includeCoordinator;
+    private final SimpleNodeSelector simpleNodeSelector;
+    private final QueryManager queryManager;
+    private final Duration estimatedExecutionTime;
+
+    public SimpleTTLNodeSelector(
+            InternalNodeManager nodeManager,
+            NodeSelectionStats nodeSelectionStats,
+            NodeTaskMap nodeTaskMap,
+            boolean includeCoordinator,
+            Supplier<NodeMap> nodeMap,
+            int minCandidates,
+            int maxSplitsPerNode,
+            int maxPendingSplitsPerTask,
+            int maxUnacknowledgedSplitsPerTask,
+            int maxTasksPerStage,
+            TTLFetcherManager ttlFetcherManager,
+            QueryManager queryManager,
+            Session session)
+    {
+        this.simpleNodeSelector = new SimpleNodeSelector(
+                nodeManager,
+                nodeSelectionStats,
+                nodeTaskMap,
+                includeCoordinator,
+                nodeMap,
+                minCandidates,
+                maxSplitsPerNode,
+                maxPendingSplitsPerTask,
+                maxUnacknowledgedSplitsPerTask,
+                maxTasksPerStage);
+
+        this.ttlFetcherManager = ttlFetcherManager;
+        this.session = session;
+        checkArgument(session.getResourceEstimates().getExecutionTime().isPresent(), "Estimated execution time is not present");
+        estimatedExecutionTime = session.getResourceEstimates().getExecutionTime().get();
+        this.nodeMap = new AtomicReference<>(nodeMap);
+        this.nodeTaskMap = nodeTaskMap;
+        this.minCandidates = minCandidates;
+        this.maxSplitsPerNode = maxSplitsPerNode;
+        this.maxPendingSplitsPerTask = maxPendingSplitsPerTask;
+        checkArgument(maxUnacknowledgedSplitsPerTask > 0, "maxUnacknowledgedSplitsPerTask must be > 0, found: %s", maxUnacknowledgedSplitsPerTask);
```

Comment:
If you follow my suggestion above on using composition to create this inner selector, it seems we can remove fields such as this.